doc/cookbook/namedRoutes/NamedRoutes.lhs

```diff
@@ -0,0 +1,373 @@
+{-# LANGUAGE DataKinds          #-}
+{-# LANGUAGE DeriveAnyClass     #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE DeriveGeneric      #-}
+{-# LANGUAGE OverloadedStrings  #-}
+{-# LANGUAGE TypeOperators      #-}
+
+import GHC.Generics               ( Generic )
+import Data.Aeson                 ( FromJSON, ToJSON )
+import Data.Proxy                 ( Proxy(..) )
+import Network.Wai.Handler.Warp   ( run )
+
+import Servant                    ( NamedRoutes
+                                  , Handler, serve )
+import Servant.API                (Capture, Delete, Get, Put, QueryParam, ReqBody
+                                  , JSON, NoContent (..)
+                                  , FromHttpApiData (..),ToHttpApiData(..)
+                                  , (:>) )
+import Servant.API.Generic        ( (:-) )
+
+import Servant.Client             ( AsClientT, ClientM, client
+                                  , (//), (/:) )
+import Servant.Client.Generic     ()
+
+import Servant.Server             ( Application, ServerT )
+import Servant.Server.Generic     ( AsServerT )
+
+```
+
+## Domain context
+
+Now that we’ve handled the boilerplate, we can dive into our Movie Catalog domain.
+
+Consider a `Movie` constructed from a `Title` and a `Year` of publication.
+
+``` haskell
+data Movie = Movie
+    { movieId :: MovieId
+    , title :: Title
+    , year :: Year
+    }
+    deriving stock Generic
+    deriving anyclass (FromJSON, ToJSON)
+
+type MovieId = String
+type Title = String
+type Year = Int
+
+```
+
+
+Let’s forget about the deriving stuff for now and think about the API that we want to make.
+
+```
+  "version" -> Get Version
+  /
+api          "list" -> Get [Movie] ?sortBy= Title | Year       (sort by the Title or the Year)
+  \        /
+  "movies"                Get Movie
+           \                /
+            Capture MovieId - Put Movie
+                            \
+                              Delete MovieId
+```
+
+In this example, we create a very simple endpoint for the Version,
+and several complex endpoints that use nested records for the CRUD part of the movie.
+
+So, the URLs would look like
+
+- GET …/version
+- GET …/movies/list?sortby=Title
+- GET …/movies/<MovieId>/
+- PUT …/movies/<MovieId>/
+- DELETE …/movies/<MovieId>
+
+### API Type
+
+Now that we have a very clear idea of the API we want to make, we need to transform it into usable Haskell code:
+
+``` haskell
+
+data API mode = API
+    { version :: mode :- "version" :> Get '[JSON] Version
+    , movies :: mode :- "movies" :> NamedRoutes MoviesAPI
+    } deriving stock Generic
+
+type Version = String -- This will do for the sake of example.
+
+```
+Here, we see the first node of our tree. It contains the two branches “version” and “movies” respectively:
+
+The “version” branch is very simple and self-explanatory.
+The “movies” branch will contain another node, represented by another record (see above). That is why we need the `NameRoutes` helper.
```

Comment:
`NamedRoutes` *

Reply:
I'd also probably suggest expanding with another sentence, along the lines of: "It allows us to reference a record-based API where a 'standard' API is expected."